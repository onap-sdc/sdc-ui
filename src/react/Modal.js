import React from 'react';
import PropTypes from 'prop-types';

import Portal from './Portal.js';
import Body from './ModalBody.js';
import Header from './ModalHeader.js';
import Footer from './ModalFooter.js';
import Title from './ModalTitle.js';

export const modalSize = {
    medium: 'md',
    large: 'l',
    extraLarge: 'xl',
    small: 'sm',
    extraSmall: 'xsm'
};

class Modal extends React.Component {
    render() {
        const { size, type, children, show } = this.props;
        return (
            <Portal>
                <div
                    ref={el => {
                        this.modalRef = el;
                    }}>
                    {show && (
                        <div className={`sdc-modal ${modalSize[size]}`}>
                            <div
                                className={`sdc-modal__wrapper sdc-modal-type-${type}`}>
                                {children}
                            </div>
                        </div>
                    )}
                    {show && <div className="modal-background" />}
                </div>
            </Portal>
        );
    }
}

Modal.defaultProps = {
    show: false,
    size: 'medium',
    type: 'info'
};

Modal.propTypes = {
    show: PropTypes.bool,
    size: PropTypes.string,
    children: PropTypes.node,
    type: PropTypes.string
};

Modal.Body = Body;
Modal.Header = Header;
Modal.Footer = Footer;
Modal.Title = Title;
export default Modal;
