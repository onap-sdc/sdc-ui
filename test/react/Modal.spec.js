import React from 'react';
import { mount } from 'enzyme';

import Modal from '../../src/react/Modal';

describe('Modal', () => {
    const MODAL_MESSAGE = 'Message';
    test('standard modal', () => {
        const modal = mount(
            <Modal show={true} size="small">
                <Modal.Header>
                    <Modal.Title>Standard Modal</Modal.Title>
                </Modal.Header>
                <Modal.Body>{MODAL_MESSAGE}</Modal.Body>
                <Modal.Footer
                    actionButtonText="Yes"
                    actionButtonClick={() => {}}
                />
            </Modal>
        );

        expect(modal.find(Modal.Body).length).toBe(1);
        expect(modal.find(Modal.Header).length).toBe(1);
        expect(modal.find(Modal.Title).length).toBe(1);
        expect(modal.find(Modal.Body).length).toBe(1);
        expect(modal.find(Modal.Footer).length).toBe(1);
        expect(modal.find(Modal.Header).props().type).toBe('info');
        expect(modal.find(Modal.Body).text()).toBe(MODAL_MESSAGE);
        expect(modal.html()).toMatchSnapshot();
    });

    test('standard modal - not displayed', () => {
        const modal = mount(
            <Modal show={false} size="small">
                <Modal.Header>
                    <Modal.Title>Standard Modal</Modal.Title>
                </Modal.Header>
                <Modal.Body>{MODAL_MESSAGE}</Modal.Body>
                <Modal.Footer
                    actionButtonText="Yes"
                    actionButtonClick={() => {}}
                />
            </Modal>
        );

        expect(modal.find(Modal.Body).length).toBe(0);
        expect(modal.html()).toMatchSnapshot();
    });

    test('alert modal', () => {
        const modal = mount(
            <Modal show type="alert" size="small">
                <Modal.Header type="alert">
                    <Modal.Title>Title</Modal.Title>
                </Modal.Header>
                <Modal.Body>{MODAL_MESSAGE}</Modal.Body>
                <Modal.Footer closeButtonText="Ok" />
            </Modal>
        );

        expect(modal.find(Modal.Body).text()).toBe(MODAL_MESSAGE);
        expect(modal.find('.sdc-modal-type-alert').length).toBe(1);
        expect(modal.html()).toMatchSnapshot();
    });

    test('custom modal', () => {
        const modal = mount(
            <Modal show type="custom">
                <Modal.Header type="custom">
                    <Modal.Title>Title</Modal.Title>
                </Modal.Header>
                <Modal.Body>{MODAL_MESSAGE}</Modal.Body>
                <Modal.Footer
                    actionButtonText="Ok"
                    actionButtonClick={() => {}}
                />
            </Modal>
        );

        expect(modal.find(Modal.Body).text()).toBe(MODAL_MESSAGE);
        expect(modal.find('.sdc-modal-type-custom').length).toBe(1);
        expect(modal.html()).toMatchSnapshot();
    });
});
