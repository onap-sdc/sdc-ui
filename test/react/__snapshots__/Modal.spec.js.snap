// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal alert modal 1`] = `"<div><div class=\\"sdc-modal sm\\"><div class=\\"sdc-modal__wrapper sdc-modal-type-alert\\"><div class=\\"sdc-alert__header sdc-modal__header\\"><div class=\\"svg-icon-wrapper sdc-modal__icon sdc-modal__svg-use   bottom\\"><svg></svg></div><div class=\\"title \\">Title</div><div class=\\"svg-icon-wrapper sdc-modal__close-button-svg sdc-modal__close-button   bottom\\"><svg></svg></div></div><div class=\\"sdc-modal__content \\">Message</div><div class=\\"sdc-modal__footer\\"><div><button class=\\"sdc-button sdc-button__primary   \\">Ok</button></div></div></div></div><div class=\\"modal-background\\"></div></div>"`;

exports[`Modal custom modal 1`] = `"<div><div class=\\"sdc-modal md\\"><div class=\\"sdc-modal__wrapper sdc-modal-type-custom\\"><div class=\\"sdc-custom__header sdc-modal__header\\"><div class=\\"title \\">Title</div><div class=\\"svg-icon-wrapper sdc-modal__close-button-svg sdc-modal__close-button   bottom\\"><svg></svg></div></div><div class=\\"sdc-modal__content \\">Message</div><div class=\\"sdc-modal__footer\\"><div><button class=\\"sdc-button sdc-button__primary   \\">Ok</button><button class=\\"sdc-button sdc-button__secondary   \\">Close</button></div></div></div></div><div class=\\"modal-background\\"></div></div>"`;

exports[`Modal standard modal - not displayed 1`] = `"<div></div>"`;

exports[`Modal standard modal 1`] = `"<div><div class=\\"sdc-modal sm\\"><div class=\\"sdc-modal__wrapper sdc-modal-type-info\\"><div class=\\"sdc-info__header sdc-modal__header\\"><div class=\\"svg-icon-wrapper sdc-modal__icon sdc-modal__svg-use   bottom\\"><svg></svg></div><div class=\\"title \\">Standard Modal</div><div class=\\"svg-icon-wrapper sdc-modal__close-button-svg sdc-modal__close-button   bottom\\"><svg></svg></div></div><div class=\\"sdc-modal__content \\">Message</div><div class=\\"sdc-modal__footer\\"><div><button class=\\"sdc-button sdc-button__primary   \\">Yes</button><button class=\\"sdc-button sdc-button__secondary   \\">Close</button></div></div></div></div><div class=\\"modal-background\\"></div></div>"`;
